### PR TITLE
fix: replace usage of intl package with custom logic

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> replace usage of intl package with custom logic

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#93]() 

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import 'package:test/test.dart';
+import 'package:rudder_sdk_flutter_platform_interface/src/utils.dart';
+
+void main() {
+  test('conversion of datestring object to string in format YY-mm-dd', () {
```
Comment (saikumarrs) [inline packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart:5]: The format is `yy-MM-dd` as per previous implementation. Have we changed it now?

File: packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    final DateTime date = DateTime(2020, 12, 12);
+    expect(Utils.toDateString(date), equals("20-12-12"));
+    final DateTime date2 = DateTime(2022, 7, 25);
+    expect(Utils.toDateString(date2), equals("22-07-25"));
+    final DateTime date3 = DateTime(2023, 1, 13);
```
Comment (saikumarrs) [inline packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart:10]: Add a case of leading zeroes for the day field too.

File: packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart (hunk 1 above)
Comment (desusai7) [reply packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart]: no, typo in the test description

Comment (desusai7) [reply packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart]: fixed it

File: packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart (hunk 2 above)
Comment (desusai7) [reply packages/plugins/rudder_plugin_interface/test/src/models/utils_test.dart:10]: added

**Review by saikumarrs — APPROVED** [review]

**Review by itsdebs — APPROVED** [review]

**Pull request merged**